### ISSUE TITLE
[Bug] Reloading .vimrc with "source %" deactivates rainbow highlighting

### ISSUE THREAD

**Issue by vedantroy** [luochen1990/rainbow#129]
Firstly, thanks for making this plugin, it's very useful!
### **Describe the bug**
When reloading the .vimrc file with `source %`. The rainbow coloring in a different split will disappear.
### **To Reproduce**
I have read the following two issues, neither of the proposed solutions worked:
https://github.com/luochen1990/rainbow/issues/33
https://github.com/luochen1990/rainbow/issues/54

I have the following code at the end of my .vimrc, but it does not solve the issue:

```
let g:rainbow_active=1
" Solution A
if exists(':RainbowToggleOn')
    "For checking if the if-statement is entered
    echom "Triggering rainbow toggle"
    exe 'RainbowToggleOn'
endif
" Solution B
call rainbow_main#load()
```

Steps:
1. Install Rainbow Parenthesis and add the following code to the end of your .vimrc. Reload your .vimrc.
2. Open a separate split with bug.cpp, the rainbow coloring should be present since the global variable `rainbow_active` is set to true.
3. Switch back to the split containing your .vimrc and then execute `:source %` to reload it.
4. The coloring in the split containing bug.cpp will have disappeared.
- Executing `:RainbowToggleOn` in the split containing the .vimrc will not re-activate the coloring in the split containing bug.cpp. Instead, you need to switch to the split containing bug.cpp and execute `:RainbowToggleOn`.

I think the problem is that `exe RainbowToggleOn` only works on the active buffer (in this case, `.vimrc`) instead of working on all buffers, so the `if` statement fails to re-active rainbow coloring in the other buffers.

**Comment by luochen1990** [comment]
You know that `source .vimrc` will only execute the commands in `.vimrc`, and the `rainbow_main` is not executed another time.  `rainbow_main` will be executed when you startup vim because it's in your `&rtp`, and at the first run, it will register `autocmd`s so that the highlighting works.  And when you execute your `.vimrc` again, somehow your `.vimrc` cleanup the `autocmd`s (maybe by a `syntax off`).

In short, resource the vimrc file is not exactly the same thing as restart your vim.  You can add the `autocmd`s to your vimrc if you insist to use the `source .vimrc` solution:

```
	auto syntax * call rainbow_main#load()
	auto colorscheme * call rainbow_main#load()
```
